### PR TITLE
Desktop theme Button

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Adapts the Button component to support the desktop theme, while maintaining backwards compatibility for the existing touchscreen themes.

_Review by commit_

## Demo Video or Screenshot
I recommend poking around in Storybook, but here's a sampling

https://github.com/votingworks/vxsuite/assets/530106/3036a9f7-db77-4b77-b0ca-084ba5bc0708

Touch screen themes

https://github.com/votingworks/vxsuite/assets/530106/e93948b5-e6e5-4412-9587-4a5c39668079

## Testing Plan
- Updated existing automated tests
- Added some new unit tests
- Visual inspection in Storybook

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/ui/src/icons.tsx [hunk 1]
```diff
@@ -104,114 +104,114 @@ export const Icons = {
...
+  ZoomIn() {
     return <FaIcon type={faMagnifyingGlassPlus} />;
   },
 
-  ZoomOut(): JSX.Element {
+  ZoomOut() {
     return <FaIcon type={faMagnifyingGlassMinus} />;
   },
-} as const;
+} satisfies Record<string, () => JSX.Element>;
```
Comment (adghayes) [inline libs/ui/src/icons.tsx:214]: You've really been using `satisfies` a lot. It would seem you're... satisfied with it.

**Review by adghayes — APPROVED** [review]
I'm really excited to see this applied to apps next! I'm enjoying the expanded button options, and am surprised how backwards compatibility continues working out with (apparent) ease.

A few UX reactions:
- As someone who doesn't distinguish color shades particularly well, I would prefer that the color change on hover is just a little more pronounced. Particularly with the "neutral" button.
- In some of the disabled styles (primary, danger, and inversePrimary) it's hard to distinguish the border because it doesn't contrast enough with the button color or, in the inversePrimary case, with the background color. I do very much like that you've preserved the dashed-outline-as-disabled convention in the desktop style. Perhaps some different border shading could help, or even just making the dashed outline a smidge thicker? Not a big deal.

File: libs/ui/src/button.tsx [hunk 2]
```diff
@@ -60,118 +75,277 @@ function resolveIcon(icon: IconName | JSX.Element): JSX.Element {
 }
 
 interface VariantConfig {
-  color: keyof UiTheme['colors'];
-  isSolidColor?: boolean;
+  color: ButtonColor;
+  fill: ButtonFill;
+  icon?: IconName | JSX.Element;
+  iconRight?: IconName | JSX.Element;
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ui/src/button.tsx:81]: Oops, thought I removed these. Must have been a rebase issue

File: libs/ui/src/button.tsx [hunk 3]
```diff
@@ -25,12 +34,18 @@ export interface StyledButtonProps {
   disabled?: boolean;
   id?: string;
   role?: string;
+  style?: React.CSSProperties;
   tabIndex?: number;
-  variant?: ButtonVariant;
+
   icon?: IconName | JSX.Element;
   rightIcon?: IconName | JSX.Element;
+  variant?: ButtonVariant;
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ui/src/button.tsx:42]: Good call, will move

File: libs/ui/src/icons.tsx (hunk 1 above)
Comment (jonahkagan) [reply libs/ui/src/icons.tsx:214]: 😼

**Comment by jonahkagan** [comment]
> * I would prefer that the color change on hover is just a little more pronounced. Particularly with the "neutral" button.

Thanks, this is helpful. I went back and forth on this one, but I'll switch to a slightly darker gray for neutral hover.

> In some of the disabled styles (primary, danger, and inversePrimary) it's hard to distinguish the border because it doesn't contrast enough with the button color

I played around with this but unfortunately using a more contrasting color makes the button looks like it has teeth on the edge rather than a dashed border. So even though it's not ideal, I think I'll leave as is for now. The disabled mouse cursor provides plenty of visual feedback IMO.

Screenshot 2023-10-31 at 8 52 31 AM

**Pull request merged**